Algorithm.CSharp/OptionGreeksRegressionAlgorithm.cs

```diff
@@ -0,0 +1,148 @@
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm reproducing issue #5610
+    /// </summary>
+    public class OptionGreeksRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private Symbol _itmCallSymbol, _otmCallSymbol, _itmPutSymbol, _otmPutSymbol;
+        private const decimal error = 0.05m;
+
+        public override void Initialize()
+        {
+            SetStartDate(2023, 8, 2);
+            SetEndDate(2023, 8, 4);
+            SetCash(1000000);
+
+            var equity = AddEquity("SPY", Resolution.Minute);
+            equity.VolatilityModel = new StandardDeviationOfReturnsVolatilityModel(30);
+
+            _itmCallSymbol = QuantConnect.Symbol.CreateOption(equity.Symbol, Market.USA, OptionStyle.American, OptionRight.Call, 430, new DateTime(2023, 9, 1));
+            _otmCallSymbol = QuantConnect.Symbol.CreateOption(equity.Symbol, Market.USA, OptionStyle.American, OptionRight.Call, 470, new DateTime(2023, 9, 1));
+            _itmPutSymbol = QuantConnect.Symbol.CreateOption(equity.Symbol, Market.USA, OptionStyle.American, OptionRight.Put, 430, new DateTime(2023, 9, 1));
+            _otmPutSymbol = QuantConnect.Symbol.CreateOption(equity.Symbol, Market.USA, OptionStyle.American, OptionRight.Put, 470, new DateTime(2023, 9, 1));
+
+            AddOptionContract(_itmCallSymbol, Resolution.Minute);
+            AddOptionContract(_otmCallSymbol, Resolution.Minute);
+            AddOptionContract(_itmPutSymbol, Resolution.Minute);
+            AddOptionContract(_otmPutSymbol, Resolution.Minute);
+        }
+
+        public override void OnData(Slice slice)
+        {
+            foreach (var kvp in slice.OptionChains)
+            {
+                var chain = kvp.Value;
+                if (chain == null)
+                {
+                    continue;
+                }
+
+                foreach (var contractKvp in chain.Contracts)
+                {
+                    var symbol = contractKvp.Key;
+                    var contract = contractKvp.Value;
+                    var delta = contract.Greeks.Delta;
+                    decimal expected;
+
+                    // Values from CBOE
+                    if (symbol == _itmCallSymbol)
+                    {
+                        expected = 0.78901m;
+                    }
+                    else if (symbol == _otmCallSymbol)
+                    {
+                        expected = 0.09627m;
+                    }
+                    else if (symbol == _itmPutSymbol)
+                    {
+                        expected = -0.18395m;
+                    }
+                    else
+                    {
+                        expected = -0.99989m;
+                    }
+
+                    if (delta >= expected + error || delta <= expected - error)
```

Comment:
You are not really asserting all the expected symbols were in the chain, think it's worth adding 👍